starport-blog/01-index.md

```diff
@@ -0,0 +1,350 @@
+
+## x/blog/types/key.go
+
+To define the post prefix add the following code:
+
+```go
+package types
+
+const (
+  // Other constants...
+  // PostPrefix is used for keys in the KV store
+  PostPrefix = "post-"
+)
+```
+
+## x/blog/types/codec.go
+
+Finally, `store.Set(key, value)` writes our post to the store.
+Two last things to do is tell our encoder how our `MsgCreatePost` is converted to bytes.
+
+```go
+func RegisterCodec(cdc *codec.Codec) {
+  cdc.RegisterConcrete(MsgCreatePost{}, "blog/CreatePost", nil)
+}
+```
+
+## x/blog/alias.go
+
+And modify `alias.go` to bring message types from types package to blog package:
+
+```go
+var (
+  // ...
+  NewMsgCreatePost = types.NewMsgCreatePost
+)
+type (
+  // ...
+  MsgCreatePost = types.MsgCreatePost
+)
+```
+
+## Launch
+
+Now we are ready to build and start our app and create some posts.
+
+To launch your application run:
+
+```
+starport serve
+```
+
+This command installs dependencies, builds and initializes the app, and runs servers. You can also do it manually:
+
+1. `go mod tidy` cleans up dependencies.
+2. `make` builds your app and creates two binaries in your go path: `blogd` and `blogcli`.
+3. Initialization scripts in the `Makefile` removes data directories, configures your app and generates two accounts. By default your app stores data in your home directory in `~/.blogd` and `~/.blogcli`. The script removes them, so every time you have a clean state.
+4. `blogd start` launches your app. After a couple of seconds you will see hashes of blocks being generated. Leave this terminal window open and open a new one.
+
+Note: depending on your OS and firewall settings, you may have to accept a prompt asking if your application's binary (`blogd` in this case) can accept external connections.
+
+Run the following command to create a post:
+
+```sh
+blogcli tx blog create-post 'This is a post!" --from=user1
+```
+
+“This is a post!” is a title for our post and `--from=user1` tells the program who is creating this post. `user1` is a label for your pair of keys used to sign the transaction, created by initialization script previously. Keys are stored in `~/.blogcli`.
+
+After running the command and confirming it, you will see an object with “txhash” property with a value like `CA1491B39384A4F29E568F62B156E0F2D0601507EF499CE1B8F3930BAFE7F03C`.
+
+To verify that the transaction has been processed, open a browser and visit the following URL (make sure to replace `CA14...` with the value of your txhash):
+
+```
+http://localhost:26657/tx?hash=0xCA1491B39384A4F29E568F62B156E0F2D0601507EF499CE1B8F3930BAFE7F03C
+```
+
+Congratulations! You have just created and launched your custom blockhain and sent the first transaction.
+
+## Errors
```

Comment:
What is this section? a FAQ? Why is it here?

Reply:
It's a list of error that developers might encounter. If they do, finding answers online may prove challenging.